pika/heartbeat.py

```diff
@@ -9,37 +9,32 @@
 
 class HeartbeatChecker(object):
     """Checks to make sure that our heartbeat is received at the expected
-    intervals.
+    timeouts.
 
     """
-    DEFAULT_INTERVAL = 60
-    MAX_IDLE_COUNT = 2
-
+    # Note: even though we're sending heartbeats in half the specified
+    # timeout value, the broker will be sending them to us at the specified
+    # value. This means we'll be checking for an idle connection via send_and_check
+    # twice as many times as the broker will send heartbeats to us,
+    # so we need to set max idle count to 4 here
+    _MAX_IDLE_COUNT = 4
+    _CONNECTION_FORCED = 320
     _STALE_CONNECTION = "Too Many Missed Heartbeats, No reply in %i seconds"
 
-    def __init__(self, connection, interval=DEFAULT_INTERVAL, idle_count=MAX_IDLE_COUNT):
+    def __init__(self, connection, timeout):
         """Create a heartbeat on connection sending a heartbeat frame every
```

Comment:
This description sounds wrong. It should send a heartbeat multiple times within the timeout window. And it will also be checking for timeout of heartbeats from the broker.